docs/concepts/floxhub-environments.md

```diff
@@ -0,0 +1,152 @@
+---
+title: "FloxHub environments"
+description: "Reusable environments that are centrally managed on FloxHub"
+---
+
+When you create an environment for your project, you often do so via the [`flox init`][flox-init] command, which creates a `.flox` directory that you can check into source control.
+
+```{ .sh .copy}
+flox init
+git add .flox
+```
+
+This allows you to track changes to the environment the same way that you track changes to your source code, but it also ties this environment to this specific project and Git repository.
+
+Another way of working is to [push][flox-push] an environment to [FloxHub][floxhub] to turn it into a reusable, centrally managed environment.
+
+```{ .sh }
+$ flox push
+✅ Updates to myenv successfully pushed to FloxHub
+
+View the environment at: https://hub.flox.dev/myuser/myenv
+Use this environment from another machine: 'flox activate -r myuser/myenv'
+Make a copy of this environment: 'flox pull myuser/myenv'
+```
+
+Once pushed, the state and history of this environment are tracked on FloxHub in the form of [generations][generations].
+Let's explore how FloxHub environments work and the kinds of workflows they enable.
+
+[flox-init]: ../man/flox-init.md
+[flox-push]: ../man/flox-push.md
+[floxhub]: ./floxhub.md
+[generations]: ./generations.md
+
+## Terminology
+
+We call an environment that has been pushed to FloxHub a **"FloxHub environment"**.
+One of the primary benefits of FloxHub environments is that they can be reused in different contexts, by different people, on different machines.
+
+This means you frequently have multiple copies of the environment:
+
+- The copy that lives on FloxHub, which is the source of truth
+- The copy that exists locally for a given user on a given machine
+
+We call the copy of the environment on FloxHub the **"upstream"** copy, and the copy on a user's machine the **"local"** copy.
+
+## Getting a FloxHub environment
+
+Once you've pushed an environment to FloxHub, you can then use the [`flox pull`][flox-pull] command to fetch a copy of the environment to your machine.
+
+A FloxHub environment can exist on your machine in two different forms:
+
+<!-- markdownlint-disable MD007 -->
+- A cached copy transparently managed by Flox
+    - Fetch this manually via `flox pull --reference <owner>/<name>`
```

Comment:
suggestion nonblocking: I think we should add that we do an implicit pull for any command that uses `--reference`